### ISSUE TITLE
Handling calls to template function

### ISSUE THREAD

**Issue by rabidbit** [google/tie#323]
User noticed the following:

Input code:

```
def reverseWords(s):
    return s

reverseWords('hi')
```

Results in the following response:

> Looks like your code had a runtime error when evaluating the input None. Here's the trace:
> 
> `NameError: name 'StudentCode' is not defined on line 4`

Input code:

```
def reverseWords(s):
    reverseWords('hi')
```

Results in the following response:

> Looks like your code had a runtime error when evaluating the input "moo cow bark dog". Here's the trace:
> 
> `ExternalError: RangeError: Maximum call stack size exceeded on line 2`

Should we consider special handling of calls to the current function like this?

**Comment by seanlip** [comment]
Discussed with @eyurko:
- For the second one, we can change the message to: "Looks like your code is hitting an infinite recursive loop. Check to see that your recursive calls terminate."
- For the first one, we can say: "Please keep your code within the existing predefined functions -- we can't process code in the global scope."

**Comment by rabidbit** [comment]
SGMT. Thanks for looking into this!

**Comment by rabidbit** [comment]
Verified UI appears to work as described (UI behavior LGTM).

**Comment by seanlip** [comment]
Fixed in #324. Thanks @dchen97!
